### PR TITLE
Updated required role for user report page GET API

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Updated required role for user report page GET API

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-59083

#### Release Note
```release-note
Fixed a bug where user with user management permission were unable to view list of users in System Console > Users page.
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]

**Comment by harshilsharma63** [comment]
@esarafianou requesting security review as this includes API permission changes.

**Review by esarafianou — APPROVED** [review]

File: server/channels/api4/report.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func (api *API) InitReports() {
 }
 
 func getUsersForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
-	if !(c.IsSystemAdmin()) {
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUserManagementUsers) {
```
Comment (M-ZubairAhmed) [inline server/channels/api4/report.go:24]: Do we have any such check in the client as well? can we cross check

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by hanzei** [comment]
@harshilsharma63 Should the check in https://github.com/mattermost/mattermost/blob/aa41c146c6a68aa268d829ac855ddc9cdad37584/server/channels/api4/report.go#L78-L80 also get updated? Looking at the permission error, it seems like it requires `PermissionSysconsoleReadUserManagementUsers`.

File: server/channels/api4/report.go (hunk 1 above)
Comment (harshilsharma63) [reply server/channels/api4/report.go:24]: No, but we have a ticket for addressing the client issue more broadly - https://mattermost.atlassian.net/browse/MM-59311

**Comment by harshilsharma63** [comment]
> @harshilsharma63 Should the check in
> 
> https://github.com/mattermost/mattermost/blob/aa41c146c6a68aa268d829ac855ddc9cdad37584/server/channels/api4/report.go#L78-L80
> 
> also get updated? Looking at the permission error, it seems like it requires `PermissionSysconsoleReadUserManagementUsers`.

No, someone with only user management permission shouldn't be allowed to export a report, it should be limited to system admins but I'll check on this with team once.

**Comment by hanzei** [comment]
Should the permission error that gets returned mention `PermissionSysconsoleReadUserManagementUsers` in this case?

File: server/channels/api4/report.go (hunk 1 above)
Comment (M-ZubairAhmed) [reply server/channels/api4/report.go:24]: and its assigned to me 🤦🏽  😆

**Pull request merged**